### DISCUSSION TITLE
Is there any official supported package team for ubuntu 22.04?

### DISCUSSION THREAD

**Discussion by ghost** [containers/podman#13097] [Q&A] [answered]
The current ubuntu 22.04 has podman 3.3.1, I guess they will stay on it.

But if people who want to keep up to date with containers/podman official release. 
it will be a problem

I wonder if there is a official supported team to package common linux distro for each new version of podman.

For reference, like nodesource, they official contribute for the nodejs distribution for each common linux distro.

https://github.com/nodesource/distributions/blob/master/README.md

![image](https://user-images.githubusercontent.com/278153/151725907-59af1e35-95c5-40b1-afe9-49b32d74ef1a.png)

If podman want to take user switch from docker. 

they should take better distro support for linux user , like what docker does.

**Comment by ghost** [comment]
The current solution, is pin on this version 3.3.1, we must using the right version to connect ubuntu server. or we will got problem.
I know the diffrent client/server podman version will cause problem. 

If we can not got the latest release, we must pin on old version.

**Comment by Luap99** [comment]
@lsm5 PTAL. Do you plan to update the kubic repos for newer releases?

**Comment by rhatdan** [comment, marked as answer]
This is not an issue, it is a discussion.

The simple answer to the question is there and `official supported package ...`?
The answer is NO. This is upstream we only support upstream, distros support packages.
We have provided kubic packages for Ubuntu in the past, but I believe we now tell people to get packages from the disto, since Podman is available via Ubuntu and Debian.

**Reply by rhatdan** [reply]
If I want to get Real Wise.  Only Supported Podman comes from Red Hat Enteprise Linux and perhaps SUSE. (Maybe Oracle Linux)

**Reply by ghost** [reply]
ok, i guess i already got your position. thanks for answer

**Comment by ghost** [comment]
因为我的母语不是英文，所以我的表达可能不是能很精准。所以我用中文再写一次。
从短视的角度来看：如果支持RHEL作为唯一的发行版，我觉得是ok的，这可以给该发行版用户带来绝佳的体验。笑傲群雄。

但是从长远角度来看，Linux是一个生态。如果你主动跟世界割裂。那么可能就一个结果：你自己玩自己的，其他开源厂商，开发者，不跟随。

随着你大权独揽，你想干什么就干什么，你可能会推出一些只有RHEL机制下才能玩转的技术特性。

这个故事听起来很熟悉。它曾经有一个名字：Docker。
Docker后来的故事大家都看到了。各大厂商，党同伐异，用各自的方案，挑战Docker公司的产品。

两败俱伤，说的就是这个现状。

其实历史上，Docker可以有机会打开心扉，开放共赢，让世界和Docker一起共襄盛举，共同进步。

但可惜历史没有如果。

我们所经历的今天，只不过是对昨日历史，拙劣的模仿。我们今日所取得的的，只不过是昨日的明日黄花。

噫，微斯人，吾谁与归。

podman， 你何去何从，你是要做RHEL的禁脔。 还是要做所有人的超级英雄。

**Comment by vahid-rms** [comment]
@rhatdan Do you mean the Podman version installed on a fresh host should be always the latest?

**Reply by rhatdan** [reply]
Nope but if something is an older version that the upstream maintainers know is fixed in a newer version then you will be told to upgrade.  We will not back port fixes to older versions. Unless it is a paid for version in something like RHEL.  In RHEL Red Hat engineers back port fixes to older versions for customers.  Even though Red Hat engineers work on the upstream we do not do this for other distributions.

**Reply by KhazAkar** [reply]
And the problem is, that ubuntu 22.04 ships with podman, but it's quite old. Either we can build podman from source to have latest, use kubic ppa to have 4.6.2 or use podman-remote-static from github release section and use VM based approach, which does not work with Podman Desktop

**Reply by rhatdan** [reply]
You need to complain to ubuntu maintainers or get someone to update kubic ppa.

**Reply by KhazAkar** [reply]
Funny thing is that only podman in Ubuntu 22.04 kubic ppa is not updated, rest is. As well as podman in debian ppa is updated. I will try to find contact to them and ask about version in ppa of podman :)

**Reply by lsm5** [reply]
@rhatdan the kubic maintainer is me :D .

@KhazAkar Like I've said in other issues, I really don't have the time to continue maintaining the kubic repo. IIRC, there's some issue about the current golang in ubuntu being too old or something. So, that means added work for me which I just can't own at this point. If someone's volunteering to co-maintain on a regular basis, I'll work with them.

**Reply by lsm5** [reply]
any case, ideal solution would be to just get ubuntu to modify its update policy or whatever so updates can land easily.

**Reply by lsm5** [reply]
oh btw, I'm told you could use the `brew` package manager to fetch podman. Apparently some people have been using it with success. Hope that helps.

**Reply by rhatdan** [reply]
Yes I know it is you, trying to get community volunteers to take care of updating it or just use brew.

**Reply by KhazAkar** [reply]
@lsm5 Oh, I didn't knew that, thanks for enlightenimg me about such issues. Seems that switching to brew would be best. But using kubic ppa is still recommended for Ubuntu on official podman website tho... How can I contribute at least that here?

**Reply by lsm5** [reply]
@KhazAkar the kubic references should be removed with https://github.com/containers/podman.io/pull/240 . @rhatdan that PR is ready for merge.

**Reply by KhazAkar** [reply]
That's great news. Only thing left is to try to find some alternative for "left-out" users after this change. Maybe podman-static from another repo?

**Comment by andrewchen5678** [comment]
Why it so difficult to provide packages for mainstream distros for podman's latest versions, I was able to vibe code a script to build the deb packages for ubuntu and debian that I actively use with just a few llm prompts https://github.com/andrewtheguy/podman-package and then one extra prompt to update the podman version when it came out. It should be easy to integrate into the current build workflow. We don't even have to provide it for those uncommon distros like those that don't use deb or rpm.
